src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileDefinition.cs

```diff
@@ -47,11 +47,6 @@ public int ShortId
 			get => _shortId ?? -1;
 			set
 			{
-				if (_shortId != null)
-				{
-					throw new InvalidOperationException($"{nameof(ShortId)} should not be set more than once.");
-				}
```

Comment:
@Youssef1313 is this still valid ?

Reply:
I think this may cause race conditions indeed. We probably should be removing this completely and use a Guid instead which is compatible with :
https://github.com/unoplatform/uno/blob/07f529ff81d03f90552d4321bd43c2b2dbab123a/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs#L1409

Reply:
I used the existing UniqueID instead of adding a Guid property, but let me know if you think a guid is better.

Reply:
It's pretty much the same at this point, it's a stable ID, only that it's a bit large. Let's take a look at what the assemblies look like with that new ID.

Reply:
It looks like this:
![image](https://user-images.githubusercontent.com/5839577/190397743-c9822f61-00c2-4b23-acd0-005d3921a911.png)

But overall, I think this line should not use any ID because it's already in a class that's defined with the same ID and that the Index is good enough to discriminate. Let's try doing this instead:

```csharp
 return $"Get_{index}"; 
```